### PR TITLE
REST Endpoint for TeamMembers

### LINKED ISSUE #16: Create an endpoint to add a new team member
Team Member Needs the following fields
Name string,
Title string,
SkillList []TMSkill,
WishList []TMSkill

### LINKED ISSUE #17: Create a GET endpoint to pull a list of TeamMembers, or a TeamMember by id
/teammembers/
/teammembers/{id}

### PR DESCRIPTION
**Pull request by Hopding**
Fixes #16 and Fixes #17

### CODE REVIEW

**Review by thebho — APPROVED** [review]

File: controller/basecontroller.go [hunk 1]
```diff
@@ -24,7 +24,8 @@ func (bc *BaseController) Init(w http.ResponseWriter, r *http.Request, session d
 // will be returned. If not, then an empty string is returned ("").
 func checkForId(url *url.URL) string {
 	base := path.Base(url.RequestURI())
-	if url.EscapedPath() != "/skills" && url.EscapedPath() != "/skills/" {
```
Comment (thebho) [inline controller/basecontroller.go:27]: Let's extract this into another function.  The logic is already starting to get difficult to read.

File: skilldirectoryschema.cql [hunk 2]
```diff
@@ -15,5 +15,12 @@ CREATE TABLE IF NOT EXISTS skills2.links (
...
 
+CREATE TABLE IF NOT EXISTS skills2.teammembers (
+    name varchar,
+    id uuid PRIMARY KEY,
+    title varchar
+);
+
 CREATE INDEX IF NOT EXISTS ON skills2.skills ( skilltype );
 CREATE INDEX IF NOT EXISTS ON skills2.links ( skill_id );
+CREATE INDEX IF NOT EXISTS ON skills2.teammembers ( title );
```
Comment (thebho) [inline skilldirectoryschema.cql]: Please remove as discussed.

**Review by thebho — CHANGES_REQUESTED** [review]

File: controller/basecontroller.go [hunk 3]
```diff
@@ -22,10 +22,33 @@ func (bc *BaseController) Init(w http.ResponseWriter, r *http.Request, session d
...
-		return base
+	if isValidEndpoint(url.EscapedPath()) {
+		return ""
 	}
-	return ""
+	return base
 }
+
+// isValidEndpoint() returns true if endpoint is an endpoint being
```
Comment (thebho) [inline controller/basecontroller.go:33]: @Viper23 Will appreciate this change

**Review by thebho — APPROVED** [review]

File: controller/basecontroller.go (hunk 3 above)
Comment (Hopding) [reply controller/basecontroller.go:33]: Who is @Viper23 ?

**Pull request merged**